cogs/study_channels.py

```diff
@@ -173,7 +173,8 @@ async def study_channel_join(self, ctx: Context, channel: int):
     @loop(minutes=10)
     async def delete_old_channels(self):
         to_delete = set()
-        for voice_channel in study_channels:
+        channels = study_channels.copy()
+        for voice_channel in channels:
             if await TmpChannelUtil.check_delete_channel(voice_channel, self.db, logger, self.bot):
                 to_delete.add(voice_channel)
         for old in to_delete:
```

Comment:
since the list is copied anyway, we don't need the `to_delete` list. Just delete immediatly in line 179.

So copy line 181 to 179 and remove the `to_delete` list.

Reply:
Okay deleted